lib/node/limits.js

```diff
@@ -0,0 +1,119 @@
+// Functions to apply limits to nodes
+
+'use strict';
+
+var nodeError = require('../limits/error');
+var estimates = require('../limits/estimates');
+
+function limitNumberOfRows(node, context, limit, callback) {
+    if (!limit.value || !isFinite(limit.value)) {
```

Comment:
The `!limit.value` check is to allow for undefined limits (and ignore them).

I'll use `Number.isFinite` instead of the global one, I wasn't aware of that.